tests/main/interfaces-network-status/task.yaml

```diff
@@ -0,0 +1,67 @@
+summary: Ensure that the network-status interface works.
+
+details: |
+    The network-status interface allows a snap to monitor network connection status.
+
+    A snap which defines the network-status plug must be shown in the interfaces list.
+    The plug is autoconnected on install and, as usual, must be able to be disconnect/reconnect.
+
+    The snap is also declaring a plug on this interface must be able to ask for its status.
+
+# dbus-launch not supported in ubuntu-core
+systems: [-ubuntu-core-16-*]
+
+prepare: |
+    . "$TESTSLIB/dbus.sh"
+    . "$TESTSLIB/dirs.sh"
+
+    echo "Given a snap declaring a plug on the network-status interface is installed"
+    snap install test-snapd-network-status-provider
+
+    echo "And the provider dbus loop is started"
+    start_dbus_unit $SNAP_MOUNT_DIR/bin/test-snapd-network-status-provider.provider
+
+restore: |
+    rm -f getstate.error
+
+    . "$TESTSLIB/dbus.sh"
+    stop_dbus_unit
+
+execute: |
+    CONNECTED_PATTERN="^test-snapd-network-status-provider:network-status-test +test-snapd-network-status-provider:network-status"
+    DISCONNECTED_PATTERN="^- +test-snapd-network-status-provider:network-status$"
+
+    echo "The interface is connected by default"
+    snap interfaces | MATCH "$CONNECTED_PATTERN"
+
+    echo "Then wait until the dbus name is properly reserved"
+    for i in $(seq 10); do
```

Comment:
If this loop doesn't match the version in 10 runs (which equals to 10 seconds, roughly) the test will carry on without an error. I'm not sure how likely that is to happen, if at all?